### PR TITLE
Migrate http-server-mock to python3

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

`http-server-mock` was the last script still in py 2.

### How?

`BaseHTTPRequestHandler` now write `bytes` instead of `string` to fd.

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [x] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

**Review by ncreated — COMMENTED** [review]
It looks ok, but for some reason server's tests for the pulling mode fail on CI:
```
Test Suite HTTPServerMockTests.xctest started
HTTPServerMockTests
    ✗ testItPullsRecordedRequests, failed - Failed to connect with the server.
    ✓ testItReturnsRecordedRequests (4.540 seconds)
    ✓ testWhenPullingRecordedRequestExceedsTimeout_itThrowsAnError (3.344 seconds)
HTTPServerMockTests.HTTPServerMockTests
  testItPullsRecordedRequests, failed - Failed to connect with the server.
  /Users/vagrant/git/instrumented-tests/http-server-mock/Tests/HTTPServerMockTests/HTTPServerMockTests.swift:59

        guard let serverProces = runner.waitUntilServerIsReachable() else {
            XCTFail("Failed to connect with the server.")
            return
```

If it's a further problem of python version vs CI machine, maybe we need to use `venv` like we do for other python tooling.

**Comment by maxep** [comment]
Yea I saw that, it runs locally tho. I will have a look!

File: instrumented-tests/http-server-mock/Sources/HTTPServerMock/ServerProcessRunner.swift [hunk 1]
```diff
@@ -22,12 +22,13 @@ public class ServerProcessRunner {
     /// Waits until server is reachable.
     /// Returns `ServerProcess` instance if the server is running, `nil` otherwise.
     public func waitUntilServerIsReachable() -> ServerProcess? {
-        let deadline = Date(timeIntervalSinceNow: 5)
+        let deadline = Date(timeIntervalSinceNow: 10)
```
Comment (maxep) [inline instrumented-tests/http-server-mock/Sources/HTTPServerMock/ServerProcessRunner.swift:25]: Looks like the first server startup takes a bit longer on CI with py3..
I've increased the deadline and wait 0.5 seconds between pings.

Comment (ncreated) [reply instrumented-tests/http-server-mock/Sources/HTTPServerMock/ServerProcessRunner.swift:25]: Let's give it a try 👍

**Review by ncreated — APPROVED** [review]
👌

**Pull request merged**